ws-client/src/client.rs

```diff
@@ -312,7 +314,21 @@ impl Client for WsClient {
 			Error::ParseError(e)
 		})?;
 		log::trace!("[frontend]: send notification: {:?}", raw);
-		let res = self.to_back.clone().send(FrontToBack::Notification(raw)).await;
+
+		let mut sender = self.to_back.clone();
+		let fut = sender.send(FrontToBack::Notification(raw));
+
+		let res = if let Some(dur) = self.request_timeout {
```

Comment:
this annoying of boiler plate but I found no good way of having a helper for `Result<T>`,  and `T` when the receiver is either expects `T` or `Result<T>`

on option is to use generic as the `http client` is using with `F: Fut -> Result<Fut::Output, Error>` but that became super complicated with lots of nested results.